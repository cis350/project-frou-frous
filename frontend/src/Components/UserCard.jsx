import React, { useState, useEffect } from 'react';
// import axios from 'axios';
import { Card, CardContent, Grid, Avatar, Typography, Button, ButtonGroup, IconButton } from '@mui/material';
import { makeStyles } from '@material-ui/core/styles';
import CloseIcon from '@mui/icons-material/Close';
import CheckIcon from '@mui/icons-material/Check';
import styled from '@mui/system/styled';
import PropTypes from 'prop-types';

// import { getTotalSkippedClasses } from 'backend/model/reportsDB';
import { getUserData, removeFriendReq, removeFriend, addFriend, sendFriendRequest, changePfp,
  getUserHistory, getUserHistoryReporter, getTotalReportHistory, getTotalClasses, getTotalReportWeek } from '../api/userPageAPI';

const LeftItem = styled('div')(() => ({
  margin: 6,
  padding: 6,
  backgroundColor: '#9ebd6e',
  borderRadius: '12px',
  textAlign: 'right',
  minHeight: 40,
}));

const RightItem = styled('div')(() => ({
  margin: 6,
  backgroundColor: '#0d1b1e',
  color: 'white',
  textAlign: 'left',
  borderRadius: '12px',
  fontSize: 26,
  minHeight: 34,
}));

const useStyles = makeStyles({
  input: {
    display: 'none',
  },
});

function UserCard(props) {
  const [profilePhoto, setProfilePhoto] = useState('');
  const [skipHistory, setSkipHistory] = useState(''); //eslint-disable-line
  const [frequentReporter, setFrequentReporter] = useState(''); //  eslint-disable-line
  const [totalReports, setTotalReports] = useState('');
  const [totalReportsWeek, setTotalReportsWeek] = useState('');
  const [totalReportsPercent, setTotalReportsPercent] = useState(''); //eslint-disable-line
  const [friendStatus, setFriendStatus] = useState('none');
  const [editingMode, setEditingMode] = useState(false);
  const [friendRequest, setFriendRequest] = useState('');
  const [newPfp, setNewPfp] = useState(null); //eslint-disable-line
  const { userId, currentUser } = props;
  const backgroundColor = '#9ebd6e';
  const classes = useStyles();

  const toBase64 = (file) => new Promise((resolve, reject) => {
    const reader = new FileReader();
    reader.readAsDataURL(file);
    reader.onload = () => resolve(reader.result);
    reader.onerror = (error) => reject(error);
  });

  const getUserInfo = async () => {
    const userData = await getUserData(userId);
    setProfilePhoto(userData.pfp);

    if (currentUser === userId) {
      setFriendStatus('currentUser');
      return;
    }

    try { // Setting friend status
      const friendData = userData.friends;
      const friendReqData = userData.friendReqs;
      if (friendData.includes(currentUser)) {
        setFriendStatus('friends');
      } else if (friendReqData.includes(currentUser)) {
        setFriendStatus('requested');
      } else {
        setFriendStatus('none');
      }
    } catch (error) {
      console.log('error', error); //eslint-disable-line    
    }
  };

  const updateFriendReq = async () => {
    try {
      const userData = await getUserData(userId);
      const { friendReqs } = userData;
      if (friendReqs.length > 0) {
        setFriendRequest(friendReqs[0]);
      } else {
        setFriendRequest('');
      }
    } catch (error) {
      console.log('error', error); //eslint-disable-line
    }
  };

  useEffect(() => {
    async function fetchUserData() {
      try {
        const skipHistoryResponse = await getUserHistory(userId);
        const frequentReporterResponse = await getUserHistoryReporter(userId);
        const totalSkips = await getTotalReportHistory(userId);
        const numClasses = await getTotalClasses(userId);
        const totalSkipsWeek = await getTotalReportWeek(userId);
        setTotalReportsPercent(typeof numClasses === 'number' ? Math.round((totalSkipsWeek / numClasses) * 100) : 'Unknown');

        if (!(skipHistoryResponse instanceof Error)) {
          setSkipHistory(skipHistoryResponse);
        }
        if (!(totalSkipsWeek instanceof Error)) {
          setTotalReportsWeek(totalSkipsWeek);
        }
        if (!(frequentReporterResponse instanceof Error)) {
          setFrequentReporter(frequentReporterResponse);
        }
        if (!(totalSkips instanceof Error)) {
          setTotalReports(totalSkips);
        }
      } catch (error) {
        console.log('error', error); //eslint-disable-line
      }
    }

    fetchUserData();
    getUserInfo();
    updateFriendReq();
  }, [userId]);

  return (
    <Card
      variant="outlined"
      sx={{
        borderRadius: '10px',
        width: '80%',
        backgroundColor: '#0d1b1e',
        border: '4px solid #E5E5E5',
        margin: 'auto',
        marginTop: '20px',
        marginBottom: '10px',
        boxShadow: '0px 0px 0px rgba(255, 255, 255, 0.3)',
        transition: 'box-shadow 0.2s ease-in-out, transform 0.2s ease-in-out',
        '&:hover': {
          boxShadow: '3px 3px 3px rgba(255, 255, 255, 0.3)', 
          transform: 'scale(1.005)',
        },
      }}
    >
      <CardContent style={{ justifyContent: 'center' }}>
        <Grid container spacing={1} sx={{ margin: '8px' }}>
          <Grid item xs={3} sm={2} style={{ padding: '8px', margin: '8px', display: 'flex', alignItems: 'center', justifyContent: 'center' }}>
            <label htmlFor="contained-button-file">
              <IconButton component="span" style={{ outline: '2px solid white', borderRadius: '50%', padding: '4px', transition: 'transform 0.3s ease-in-out' }}>
                <Avatar
                  src={newPfp ? URL.createObjectURL(newPfp) : profilePhoto}
                  alt={userId}
                  sx={{
                    width: 80,
                    height: 80,
                    cursor: editingMode ? 'pointer' : 'default',
                    transition: 'transform 0.3s ease-in-out',
                    '&:hover': {
                      transform: 'scale(1.05)',
                    },
                  }}
                />
              </IconButton>
              {editingMode && (
                <input
                  accept="image/*"
                  className={classes.input}
                  id="contained-button-file"
                  type="file"
                  onChange={(event) => {
                    setNewPfp(event.target.files[0]);
                  }}
                />
              )}
            </label>
          </Grid>
          <Grid item xs={3} sm={4} style={{ padding: '8px', margin: '8px', display: 'flex', alignItems: 'center', justifyContent: 'flex-start' }}>
            <div style={{ textAlign: 'left', color: 'white', fontSize: '2em', fontWeight: 'bold' }}>
              {userId}
            </div>
          </Grid>

          <Grid item xs={4} sm={4} style={{ padding: '8px', margin: '8px', display: 'flex', alignItems: 'center', justifyContent: 'flex-end' }}>
            {friendStatus === 'currentUser' && (
            <div style={{ justifyContent: 'center', display: 'flex', flexDirection: 'column', alignItems: 'center' }}>
              <Button
                variant="contained"
                sx={{
                  backgroundColor,
                  color: 'white',
                  marginTop: '5%',
                  borderRadius: '15px',
                  transition: 'background-color 0.2s ease-in-out, transform 0.2s ease-in-out',
                  '&:hover': {
                    backgroundColor: '#397367',
                  },
                  '&:active': {
                    transform: 'scale(0.98)',
                  },
                }}
                onClick={async () => {
                  if (newPfp !== null && editingMode) {
                    const imageBase = await toBase64(newPfp);
                    changePfp(currentUser, imageBase);
                  }
                  setEditingMode((prevEditingMode) => !prevEditingMode);
                }}
              >
                {editingMode ? 'Save' : 'Edit Picture'}
              </Button>
              {friendRequest !== '' && (
              <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center' }}>
                <ButtonGroup variant="contained" data-testid="fr" size="small" aria-label="friendRequest" sx={{ margin: '8px', marginBottom: '0px' }}>
                  <Button
                    onClick={async () => {
                      await removeFriendReq(userId, friendRequest);
                      updateFriendReq();
                    }}
                    style={{
                      backgroundColor: '#904C77',
                      color: 'white',
                      '&:active': {
                        transform: 'scale(0.98)',
                      },
                    }}
                  >
                    <CloseIcon />
                  </Button>
                  <Button
                    onClick={async () => {
                      await addFriend(userId, friendRequest);
                      updateFriendReq();
                    }}
                    style={{
                      backgroundColor: '#397367',
                      color: 'white',
                      '&:active': {
                        transform: 'scale(0.98)',
                      },
                    }}
                  >
                    <CheckIcon />
                  </Button>
                </ButtonGroup>
                <div style={{ color: 'white', paddingTop: '0px', marginTop: '0px' }}>
                  Friend Request: {friendRequest}
                </div>
              </div>            
              )}
            </div>
            )}

            {friendStatus === 'friends' && (
            <Button
              variant="contained"
              sx={{
                backgroundColor,
                color: 'white',
                marginTop: '5%',
                borderRadius: '15px',
                transition: 'background-color 0.2s ease-in-out, transform 0.2s ease-in-out',
                '&:hover': {
                  backgroundColor: '#397367',
                },
                '&:active': {
                  transform: 'scale(0.98)',
                },
              }}
              onClick={() => {
                removeFriend(userId, currentUser);
                setFriendStatus('none');
              }}
            >
              Remove Friend
            </Button>
            )}

            {friendStatus === 'requested' && (
            <Button
              variant="contained"
              sx={{
                backgroundColor,
                color: 'white',
                marginTop: '5%',
                borderRadius: '15px',
                transition: 'background-color 0.2s ease-in-out, transform 0.2s ease-in-out',
                '&:hover': {
                  backgroundColor: '#397367',
                },
                '&:active': {
                  transform: 'scale(0.98)',
                },
              }}
              onClick={() => {
                removeFriendReq(currentUser, userId);
                setFriendStatus('none');
              }}
            >
              Requested
            </Button>
            )}
            {friendStatus === 'none' && (
            <Button
              variant="contained"
              sx={{
                backgroundColor,
                color: 'white',
                marginTop: '5%',
                borderRadius: '15px',
                transition: 'background-color 0.2s ease-in-out, transform 0.2s ease-in-out',
                '&:hover': {
                  backgroundColor: '#397367',
                },
                '&:active': {
                  transform: 'scale(0.98)',
                },
              }}
              onClick={() => {
                sendFriendRequest(currentUser, userId);
                setFriendStatus('requested');
              }}
            >
              Add Friend
            </Button>
            )}

          </Grid>
        </Grid>
        <Grid container spacing={2} style={{ padding: 0, margin: 0}}>
          <Grid item style={{ paddingTop: 0, marginTop: 0}}>
            <Typography
              variant="h5"
              sx={{
                color: 'white',
                fontFamily: 'Open Sans, sans-serif',
                fontSize: '12x',
                textAlign: 'center',
                fontWeight: '600',
                fontStyle: 'italic',
                padding: '0px',
              }}
            >
              Your Weekly Report:
            </Typography>
          </Grid>
        </Grid>

        <Grid container spacing={2}>
          <Grid item xs={8} >
            <LeftItem style={{ textAlign: 'center' }}>
              Last Reporter:
            </LeftItem>
          </Grid>
          <Grid item xs={4} >
            <RightItem>
              {skipHistory || 'No Reports Yet'}
            </RightItem>
          </Grid>
        </Grid>


        <Grid container spacing={2}>
          <Grid item xs={8}>
            <LeftItem style={{ textAlign: 'center' }}> Total Reports Overall: </LeftItem>
          </Grid>
          <Grid item xs={4}>
            <RightItem>
              {totalReports || 0}
            </RightItem>
          </Grid>
        </Grid>

        <Grid container spacing={2}>
          <Grid item xs={8}>
            <LeftItem style={{ textAlign: 'center' }}> Total Reports This Week: </LeftItem>
          </Grid>
          <Grid item xs={4}>
            <RightItem>
              {totalReportsWeek || 0}
            </RightItem>
          </Grid>
        </Grid>

        <Grid container spacing={2}>
          <Grid item xs={8}>
            <LeftItem style={{ textAlign: 'center' }}> Percentage Skipped This Week: </LeftItem>
          </Grid>
          <Grid item xs={4}>
            <RightItem>
              {totalReportsPercent || 0}
              %
            </RightItem>
          </Grid>
        </Grid>

        <Grid container spacing={2}>
          <Grid item xs={8}>
            <LeftItem style={{ textAlign: 'center' }}> Most Frequent Reporter: </LeftItem>
          </Grid>
          <Grid item xs={4}>
            <RightItem>
              {frequentReporter || 'Unknown'}
            </RightItem>
          </Grid>
        </Grid>

      </CardContent>
    </Card>
  );
}

UserCard.propTypes = {
  userId: PropTypes.string.isRequired,
  currentUser: PropTypes.string.isRequired,
};

export default UserCard;
