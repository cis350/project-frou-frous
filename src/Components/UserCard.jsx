import React, { useState, useEffect } from 'react';
// import axios from 'axios';
import { Card, CardContent, Grid, Avatar, Typography, Button, ButtonGroup, IconButton } from '@mui/material';
import { makeStyles } from '@material-ui/core/styles';
import CloseIcon from '@mui/icons-material/Close';
import CheckIcon from '@mui/icons-material/Check';
import styled from '@mui/system/styled';
import PropTypes from 'prop-types';

import { getUserHistory, getUserData, removeFriendReq, removeFriend, addFriend, sendFriendRequest, changePfp } from '../api/userPageAPI';

const LeftItem = styled('div')(() => ({
  margin: 6,
  padding: 8,
  backgroundColor: '#9ebd6e',
  borderRadius: '12px',
  textAlign: 'right',
  minHeight: 40,
}));

const RightItem = styled('div')(() => ({
  margin: 6,
  padding: 12,
  backgroundColor: '#0d1b1e',
  color: 'white',
  textAlign: 'left',
  borderRadius: '12px',
  fontSize: 26,
  minHeight: 34,
}));

const useStyles = makeStyles({
  input: {
    display: 'none',
  },
});

function UserCard(props) {
  const [profilePhoto, setProfilePhoto] = useState('');
  const [skipHistory, setSkipHistory] = useState(''); //eslint-disable-line
  const [totalClasses, setTotalClasses] = useState(''); //  eslint-disable-line
  const [friendStatus, setFriendStatus] = useState('none');
  const [editingMode, setEditingMode] = useState(false);
  const [friendRequest, setFriendRequest] = useState('');
  const [newPfp, setNewPfp] = useState(null); //eslint-disable-line
  const { userId, currentUser } = props;
  const backgroundColor = '#9ebd6e';
  const classes = useStyles();

  const toBase64 = (file) => new Promise((resolve, reject) => {
    const reader = new FileReader();
    reader.readAsDataURL(file);
    reader.onload = () => resolve(reader.result);
    reader.onerror = (error) => reject(error);
  });

  const getUserInfo = async () => {
    const userData = await getUserData(userId);
    setProfilePhoto(userData.pfp);

    if (currentUser === userId) {
      setFriendStatus('currentUser');
      return;
    }

    try { // Setting friend status
      const friendData = userData.friends;
      const friendReqData = userData.friendReqs;
      if (friendData.includes(currentUser)) {
        setFriendStatus('friends');
      } else if (friendReqData.includes(currentUser)) {
        setFriendStatus('requested');
      } else {
        setFriendStatus('none');
      }
    } catch (error) {
      console.log('error', error); //eslint-disable-line    
    }
  };

  const updateFriendReq = async () => {
    try {
      const userData = await getUserData(userId);
      const { friendReqs } = userData;
      if (friendReqs.length > 0) {
        setFriendRequest(friendReqs[0]);
      } else {
        setFriendRequest('');
      }
    } catch (error) {
      console.log('error', error); //eslint-disable-line
    }
  };

  useEffect(() => {
    async function fetchUserData() {
      try {
        const reportResponse = await getUserHistory(userId);
        setSkipHistory(reportResponse.skipHistory);
        setTotalClasses(reportResponse.classes);
      } catch (error) {
        console.log('error', error); //eslint-disable-line
      }
    }

    fetchUserData();
    getUserInfo();
    updateFriendReq();
  }, [userId]);

  return (
    <Card
      variant="outlined"
      sx={{
        borderRadius: '10px',
        width: '80%',
        backgroundColor: '#0d1b1e',
        border: '3px solid #E5E5E5',
        margin: 'auto',
        marginTop: '20px',
      }}
    >
      <CardContent>
        <Grid container spacing={2} sx={{ margin: 'auto' }}>
          <Grid item sm={3} md={3}>
            <label htmlFor="contained-button-file">
              <IconButton component="span">
                <Avatar
                  src={newPfp ? URL.createObjectURL(newPfp) : profilePhoto}
                  alt={userId}
                  sx={{
                    width: 70,
                    height: 70,
                    cursor: editingMode ? 'pointer' : 'default',
                  }}
                />
              </IconButton>
              {editingMode && (
              <input
                accept="image/*"
                className={classes.input}
                id="contained-button-file"
                type="file"
                onChange={(event) => {
                  setNewPfp(event.target.files[0]);
                }}
              />
              )}
            </label>
          </Grid>
          <Grid item sm={9} md={3}>
            <div
              style={{ textAlign: 'left', marginTop: '15%', color: 'white', fontSize: '1.5em' }}
            >
              {userId}
            </div>
          </Grid>
          <Grid item sm={12} md={6}>
            {friendStatus === 'currentUser' && (
            <div>
              <Button
                variant="contained"
                sx={{ backgroundColor, color: 'white', marginTop: '5%', borderRadius: '15px' }}
                onClick={async () => {
                  if (newPfp !== null && editingMode) {
                    const imageBase = await toBase64(newPfp);
                    console.log(imageBase, 'here');
                    console.log(changePfp(currentUser, imageBase));
                  }
                  setEditingMode((prevEditingMode) => !prevEditingMode);
                }}
              >
                {editingMode ? 'Save' : 'Edit Profile'}
              </Button>
              {friendRequest !== '' && (
              <ButtonGroup variant="contained" data-testid="fr" size="small" aria-label="friendRequest">
                <Button
                  onClick={async () => {
                    await removeFriendReq(userId, friendRequest);
                    updateFriendReq();
                  }}
                  style={{ backgroundColor: '#904C77', color: 'white' }}
                >
                  <CloseIcon />
                </Button>
                <Button
                  onClick={async () => {
                    await addFriend(userId, friendRequest);
                    updateFriendReq();
                  }}
                  style={{ backgroundColor: '#397367', color: 'white' }}
                >
                  <CheckIcon />
                </Button>
                <div style={{ color: 'white' }}>
                  Friend Request from:
                  {' '}
                  {friendRequest}
                </div>
              </ButtonGroup>
              )}
            </div>
            )}

            {friendStatus === 'friends' && (
            <Button
              variant="contained"
              sx={{ backgroundColor, color: 'white', marginTop: '5%', borderRadius: '15px' }}
              onClick={() => {
                removeFriend(userId, currentUser);
                setFriendStatus('none');
              }}
            >
              Remove Friend
            </Button>
            )}

            {friendStatus === 'requested' && (
            <Button
              variant="contained"
              sx={{ backgroundColor: '#97B96B', color: 'white', marginTop: '5%', borderRadius: '15px' }}
              onClick={() => {
                removeFriendReq(currentUser, userId);
                setFriendStatus('none');
              }}
            >
              Requested
            </Button>
            )}
            {friendStatus === 'none' && (
            <Button
              variant="contained"
              sx={{ backgroundColor, color: 'white', marginTop: '5%', borderRadius: '15px' }}
              onClick={() => {
                sendFriendRequest(currentUser, userId);
                setFriendStatus('requested');
              }}
            >
              Add Friend
            </Button>
            )}

          </Grid>
        </Grid>
        <Grid container spacing={2} sx={{ margin: 'auto' }}>
          <Grid item>
            <Typography
              variant="h5"
              sx={{
                color: 'white',
                fontFamily: 'Open Sans, sans-serif',
                fontSize: '12x',
                textAlign: 'center',
              }}
            >
              Your Weekly Report:
            </Typography>
          </Grid>
        </Grid>
        <Grid container spacing={2}>
          <Grid item xs={6}>
            <LeftItem style={{ textAlign: 'center' }}>
              Total Classes Skipped:
            </LeftItem>
          </Grid>
          <Grid item xs={6}>
            <RightItem>5</RightItem>
          </Grid>
        </Grid>

        <Grid container spacing={2}>
          <Grid item xs={6}>
            <LeftItem style={{ textAlign: 'center' }}> Percent Classes Skipped: </LeftItem>
          </Grid>
          <Grid item xs={6}>
            <RightItem>40%</RightItem>
          </Grid>
        </Grid>

        <Grid container spacing={2}>
          <Grid item xs={6}>
            <LeftItem style={{ textAlign: 'center' }}> Class Most Often Skipped: </LeftItem>
          </Grid>
          <Grid item xs={6}>
            <RightItem>XYZ 101</RightItem>
          </Grid>
        </Grid>

        <Grid container spacing={2}>
          <Grid item xs={6}>
            <LeftItem style={{ textAlign: 'center' }}> Class Least Often Skipped: </LeftItem>
          </Grid>
          <Grid item xs={6}>
            <RightItem> ABC 201</RightItem>
          </Grid>
        </Grid>
      </CardContent>
    </Card>
  );
}

UserCard.propTypes = {
  userId: PropTypes.string.isRequired,
  currentUser: PropTypes.string.isRequired,
};

export default UserCard;
