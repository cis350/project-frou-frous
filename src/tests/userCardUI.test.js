/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom/extend-expect';
// import testing library functions
import { render, screen } from '@testing-library/react';
import renderer, { act } from 'react-test-renderer';
import UserCard from '../Components/UserCard';

test('rendersAllStatBoxes', async () => {
  const { getByText } = await act(async () => render(<UserCard userId={12345} />)); // eslint-disable-line
  expect(screen.getByText(/Total Classes Skipped/)).toBeInTheDocument();
  expect(screen.getByText(/Percent Classes Skipped/)).toBeInTheDocument();
  expect(screen.getByText(/Class Most Often Skipped/)).toBeInTheDocument();
  expect(screen.getByText(/Class Least Often Skipped/)).toBeInTheDocument();
});

test('rendersAvatar', async () => {
  const { getByText } = await act(async () => render(<UserCard userId={12345} />));
  const username = getByText(/MockUsername/);
  expect(username).toBeInTheDocument();
});

test('snapshot test', () => {
  const component = renderer.create(<UserCard userId={12345} />);
  const domTreeJSON = component.toJSON();

  expect(domTreeJSON).toMatchSnapshot();
});
