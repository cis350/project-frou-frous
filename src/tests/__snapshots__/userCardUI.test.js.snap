// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-zuo7fl-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1ftwddu-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-3 css-uk9mo7-MuiGrid-root"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1wnqcao-MuiAvatar-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-8 MuiGrid-grid-md-3 css-y1wd51-MuiGrid-root"
      >
        <div
          contentEditable={false}
          style={
            {
              "color": "white",
              "fontSize": "1.5em",
              "marginTop": "5%",
              "textAlign": "left",
            }
          }
          suppressContentEditableWarning={true}
        >
          jess
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6 css-vexj24-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-vn8m6k-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          Add Friend
        </button>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1ftwddu-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 css-jr0pcx-MuiTypography-root"
        >
          Your Weekly Report:
        </h5>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          className="css-zbvxyy"
          style={
            {
              "textAlign": "center",
            }
          }
        >
          Total Classes Skipped:
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          className="css-ejmnzv"
        >
          5
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          className="css-zbvxyy"
          style={
            {
              "textAlign": "center",
            }
          }
        >
           Percent Classes Skipped: 
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          className="css-ejmnzv"
        >
          40%
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          className="css-zbvxyy"
          style={
            {
              "textAlign": "center",
            }
          }
        >
           Class Most Often Skipped: 
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          className="css-ejmnzv"
        >
          XYZ 101
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          className="css-zbvxyy"
          style={
            {
              "textAlign": "center",
            }
          }
        >
           Class Least Often Skipped: 
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          className="css-ejmnzv"
        >
           ABC 201
        </div>
      </div>
    </div>
  </div>
</div>
`;
